src/main/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialect.java

```diff
@@ -0,0 +1,1753 @@
+        if (optional) {
+          builder.field(fieldName, Schema.OPTIONAL_STRING_SCHEMA);
+        } else {
+          builder.field(fieldName, Schema.STRING_SCHEMA);
+        }
+        break;
+      }
+
+      // Binary == fixed bytes
+      // BLOB, VARBINARY, LONGVARBINARY == bytes
+      case Types.BINARY:
+      case Types.BLOB:
+      case Types.VARBINARY:
+      case Types.LONGVARBINARY: {
+        if (optional) {
+          builder.field(fieldName, Schema.OPTIONAL_BYTES_SCHEMA);
+        } else {
+          builder.field(fieldName, Schema.BYTES_SCHEMA);
+        }
+        break;
+      }
+
+      // Date is day + moth + year
+      case Types.DATE: {
+        SchemaBuilder dateSchemaBuilder = Date.builder();
+        if (optional) {
+          dateSchemaBuilder.optional();
+        }
+        builder.field(fieldName, dateSchemaBuilder.build());
+        break;
+      }
+
+      // Time is a time of day -- hour, minute, seconds, nanoseconds
+      case Types.TIME: {
+        SchemaBuilder timeSchemaBuilder = Time.builder();
+        if (optional) {
+          timeSchemaBuilder.optional();
+        }
+        builder.field(fieldName, timeSchemaBuilder.build());
+        break;
+      }
+
+      // Timestamp is a date + time
+      case Types.TIMESTAMP: {
+        SchemaBuilder tsSchemaBuilder = org.apache.kafka.connect.data.Timestamp.builder();
+        if (optional) {
+          tsSchemaBuilder.optional();
+        }
+        builder.field(fieldName, tsSchemaBuilder.build());
+        break;
+      }
+
+      case Types.ARRAY:
+      case Types.JAVA_OBJECT:
+      case Types.OTHER:
+      case Types.DISTINCT:
+      case Types.STRUCT:
+      case Types.REF:
+      case Types.ROWID:
+      default: {
+        log.warn("JDBC type {} ({}) not currently supported", sqlType, columnDefn.typeName());
+        return null;
+      }
+    }
+    return fieldName;
+  }
+
+  @Override
+  public void applyDdlStatements(
+      Connection connection,
+      List<String> statements
+  ) throws SQLException {
+    try (Statement statement = connection.createStatement()) {
+      for (String ddlStatement : statements) {
+        statement.executeUpdate(ddlStatement);
+      }
+    }
+  }
+
+  @Override
+  public ColumnConverter createColumnConverter(
+      ColumnMapping mapping
+  ) {
+    return columnConverterFor(mapping, mapping.columnDefn(), mapping.columnNumber(),
+                              jdbcDriverInfo().jdbcVersionAtLeast(4, 0)
+    );
+  }
+
+  protected ColumnConverter columnConverterFor(
+      final ColumnMapping mapping,
+      final ColumnDefinition defn,
+      final int col,
+      final boolean isJdbc4
+  ) {
+    switch (mapping.columnDefn().type()) {
+
+      case Types.BOOLEAN: {
+        return new ColumnConverter() {
+          @Override
+          public Object convert(ResultSet rs) throws SQLException {
+            return rs.getBoolean(col);
+          }
+        };
+      }
```

Comment:
Okay, I debated about going this route of returning a function `ColumnConverter` that can convert a column value into an object used in a source record, versus just having this method do the conversion and return the value directly. Dealing with functions in Java 7 is verbose, but in Java 8 nearly all of this boilerplate will go away. For example, the above block in Java 8 can be:

```
case Types.BOOLEAN: {
  return (rs)->rs.getBoolean(col);
}
```
Now, the benefit of returning a function is that any overhead of looking at the column definition or whether the driver supports JDBC 4 can be done once, and the resulting function can be called many times (once per record) without this overhead.

So, I decided that returning a function is (or rather _can be_) significantly more efficient in some cases, and it's worth the short-term verbosity for the next 6 months.